pkg/pxc/backup/job.go

```diff
@@ -183,6 +202,24 @@ func (Backup) SetStorageS3(job *batchv1.JobSpec, cr *api.PerconaXtraDBCluster, s
 	}
 	job.Template.Spec.Containers[0].Env = append(job.Template.Spec.Containers[0].Env, bucket, bucketPath)
 
+	if cr.Spec.Backup.Resources != nil {
+		job.Template.Spec.Containers[0].Resources = *cr.Spec.Backup.Resources
+	}
+	if len(cr.Spec.Backup.Tolerations) > 0 {
+		job.Template.Spec.Tolerations = cr.Spec.Backup.Tolerations
+	}
+	if cr.Spec.Backup.Affinity.Advanced != nil {
+		job.Template.Spec.Affinity = cr.Spec.Backup.Affinity.Advanced
+	}
+	if len(cr.Spec.Backup.Labels) > 0 {
+		job.Template.Labels = cr.Spec.Backup.Labels
+	}
+	if len(cr.Spec.Backup.Annotations) > 0 {
+		job.Template.Annotations = cr.Spec.Backup.Annotations
+	}
+	job.Template.Spec.SchedulerName = cr.Spec.Backup.SchedulerName
+	job.Template.Spec.PriorityClassName = cr.Spec.Backup.PriorityClassName
+
```

Comment:
Why this code is part of the storage setup?

Reply:
each storage can be accessible from own nodes (like slow HDD for weekly backup is connected to node1, and fast SSD to node2) -- so "Affinity" definitely part of storage.
unfortunately, some storages (like portworx) require own Scheduler -- so "SchedulerName" also part of storage.
sometimes users can use "Tails" to avoid randomly assigned Pod on certain nodes (like on nodes with storage) -- so "Tolerations" part of the storage section.
streaming to S3 is a memory consuming operation, but storing to the disk is not -- so "Resources" part of the storage section.
Sometimes, some features (like scheduler behaviour) can be modified via annotations -- so "Annotations" part of the storage section.